### ISSUE TITLE
.gitignore: .aider* -> .aider.* to preserve .aiderignore by default.

### ISSUE THREAD

**Issue by zackees** [paul-gauthier/aider#471]
When asking questions or reporting issues, it is very helpful if you can include:

```
Starting aider with model gpt-4

Loading aider:
  remember to use /help for a list of commands

Aider v0.23.0
VSCode terminal detected, pretty output has been disabled.
Add .aider* to .gitignore (recommended)? n
Model: gpt-4 using diff edit format
Git repo: .git with 16,518 files
Warning: For large repos, consider using an .aiderignore file to ignore irrelevant files/dirs.
Repo-map: using 1024 tokens
Added Dockerfile to the chat.
```

**Comment by paul-gauthier** [comment]
I'm not sure I understand what you're trying to ask/suggest/report?

**Comment by zackees** [comment]
I'd like to keep .aiderignore because it prevents a large repo from causing aider to freeze. This needs to be checked in so that others can use aider as well. However your rule prevents this because .aider* will ignore .aiderignore. To solve this I added .aider.* so that just the other files are ignored.

**Comment by paul-gauthier** [comment]
You just need to manually manage/edit the .gitignore file for your project. You don't need to change aider.

**Comment by zackees** [comment]
I did edit the .gitignore manually. The bug is that aider keeps bugging me to add the pattern exactly as it's hardcoded in the project. So simplest solution seems to be to change the hardcoded pattern.

**Comment by paul-gauthier** [comment]
Once you've added `.aiderignore` to `.gitignore` it doesn't matter if `.aider.*` is in `.gitignore` after that. The `.aiderignore` file is already part of git.

```
tmp$ mkdir ignore
tmp$ cd ignore
tmp/ignore$ echo one > .aiderignore
tmp/ignore$ git init
Initialized empty Git repository in /Users/gauthier/tmp/ignore/.git/
tmp/ignore$ git add .aiderignore
tmp/ignore$ gc -m initial
[main (root-commit) 951a16e] initial
 1 file changed, 1 insertion(+)
 create mode 100644 .aiderignore
tmp/ignore$ echo .aiderignore > .gitignore
tmp/ignore$ echo two >> .aiderignore
tmp/ignore$ git status
On branch main
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   .aiderignore

no changes added to commit (use "git add" and/or "git commit -a")
tmp/ignore$ gc -a -m .aiderignore
[main 486233c] .aiderignore
1 file changed, 1 insertion(+)
```

**Comment by zackees** [comment]
Okay, but why-by-default try to exclude the .aiderignore file? All the rest make sense. You don't want to add in the chat logs.

Let me give you an example:

In my current project, we migrated an EC2 instance to docker and we over-included files to make it work. Aider just halts.

I struggled a little to add the aiderignore because the tool just wanted to revert it. However by default, my front end is passing no auto commit, so I might have a different default experience than you.

But anyway, just wanted to help a small friction point I experienced. Its up to you to accept the PR or not, since it's your project. Great job. My bill to openai went up a ton too but it's def worth it!!
